design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,236 @@
+
+Filter options will be determined by the data type of the input column.
+
+Filters can be added by clicking on the `Add Filter` option from the query configuration panel or directly from the results table column header menu.
+
+Columns with filters applied will display a filter icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Input Filter](https://share.balsamiq.com/c/tK2hZy6FB4zVYpN56ejpBk.png)
+
+### 2.5 Aggregating Input Column Values
+
+In cases where a link may refer to multiple values, input columns can be aggregated. In such cases, the system will perform automatic aggregation based on the data type of the referenced column. Users can change the type of aggregation at any time.
+
+Aggregated columns will have an aggregation icon indicator in the column header.
+
+---
+Wireframes
+
+[Added Input Aggregation](https://share.balsamiq.com/c/fNcbvQ1q5xMyni5JxBg3Rc.png)
+
+### 2.6 Applying a Formula to an Input Column
+
+Direct input columns can also be transformed into formulas. The available formulas will depend on the input column type.
+
+To transform the input column into a formula, the user will select the `Apply Formula` option in the column header menu.
+
+---
+Wireframes
+
+[Applying a Formula](https://share.balsamiq.com/c/nEzYaNKNTSv9EFwzwtfSof.png)
+
+## 3. Applying Transformations to the Result Table
+
+The result table refers to the resulting table from all input columns selected and added, including filters and aggregations. Transforming this output can be done by applying output filters, sorts, and summarizations to the result table.
+
+### 3.1. Filtering the Output Table
+
+Users can filter the output table by adding a filter step to any column from the result table. The column selector, in this scenario, will only allow users to select input columns rather than the whole column list from the column selection stage.
+
+---
+Wireframes
+
+[Output Filter](https://share.balsamiq.com/c/dAVYnp8VnG2r2HzkSZ3rgp.png)
+
+### 3.2. Summarizing the Result Table
+
+The `Summarization` step enables users to generate a summarized version of the result table. When added, the system will group the table values based on a summary column of their choice. The summary can be done using the summary column's exact values or by using the summary column's data type's available grouping options.
+
+To summarize the table, the user selects the `Summarize` option from the result transformations menu and adds the step. Once added, the remaining columns are automatically assigned an aggregation function, which users can change at any time. The data type of the output column is used to infer aggregates.
+
+---
+Wireframes
+
+[Output Summarization](https://share.balsamiq.com/c/rPMwwETnuQ8a4ut8NfmcDB.png)
+
+[Summarization Options](https://share.balsamiq.com/c/x2iwVBT3NwzmKyhnxSiJku.png)
+
+### 3.3 Sorting the Output Table
+
+Users can sort the output table by applying a sort to any of the result table columns.
```

Comment:
Does this include other aggregations? I'm not sure how this should work for non-numeric aggregations.

Reply:
For the time being, non-numeric aggregations can only be displayed as lists.

Reply:
My question in that case is whether we want to sort by lists. That may be an awkward concept for users, since sorting by a list isn't a very natural thing to do, and there are a number of slightly weird options for how the sorting would actually work.

Reply:
Maybe we don't allow users to sort by aggregated columns? I can see how that would get confusing.

@ghislaineguerin what do you think?